### DISCUSSION TITLE
Collaboration with GFDL

### DISCUSSION THREAD

**Discussion by chiaweh2** [NOAA-CEFI-Portal/regional_mom6#20] [General]
@yichengt900 @andrew-c-ross @KevinCounts 

We are working on a regional_mom6 module started for internal use but have the goal of providing to external user in the future. This package is also aiming to allow people to access the data that will be provided through CEFI. Currently we have plan to provide the OPeNDAP access through PSL server. We also have plan to host more data on AWS S3 through NOAA Open Data Dissemination.

Current plan is that we will start from including the existing script that Andrew and You have made available in https://github.com/NOAA-GFDL/CEFI-regional-MOM6/tree/main/diagnostics and make it more modular which is easy to apply regardless of the data source (other reanalysis data, too). So unlike the CEFI-regional-MOM6 repo, this package will be focus heavily on data IO and the post-processing and analyses that will be performed on simulation output. We have also considered to include more analyses that is not included in https://github.com/NOAA-GFDL/CEFI-regional-MOM6/tree/main/diagnostics. 

I think a discussion on this which can potentially link the two repo is a good idea. Please feel free to chime in on any input or things that you want to discuss.

**Comment by yichengt900** [comment]
@chiaweh2

Thanks. I think creating a generic Python module for analyzing data provided through CEFI is a great idea.

Although the scope of the Python module differs slightly from our work on CEFI-regional-MOM6, which is more focused on modelers directly analyzing model results, I can still see the potential in integrating the Python module for CEFI data analysis into the CEFI-regional-MOM6 workflow once it becomes more mature and generic. This integration would allow scientists and modelers to use the same modules for analyzing model output across different domains. I think regular synchronization between two repositories, especially for the diagnostic tools, would be beneficial to ensure we're aligned and making the most of each other's work.

I am also glad to know that the NWA scripts are useful and are your starting point. We're also actively developing scripts for other domains, like the North East Pacific, which will be available soon.

**Reply by chiaweh2** [reply]
Thank you @yichengt900! It is definitely very encouraging to hear that you also see the possible value of having the package. We will keep on the work and make sure the two teams sync on a relatively regular basis! Thanks again for the input and please feel free to open any issue or PR if you also want to contribute on the effort. Also, please feel free to tag us if the new domain is ready so we can also include those into the package.

**Comment by charliestock** [comment]
@chiaweh2, thanks for sharing this effort.  I look forward to follow on discussion along the lines of what you and @yichengt900 have suggested.  Hope all is well!

**Reply by chiaweh2** [reply]
@charliestock I will definitely keep both teams updated on the effort!
